File: Keepr/Models/Vault.cs
namespace Keepr.Models;

public class Vault
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public string Name { get; set; }
  public string Description {get; set;}
  public string Img { get; set; }
  public bool IsPrivate { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: Keepr.client/src/models/Vault.js
import { Profile } from "./Account.js"

export class Vault {
  constructor(data) {
    this.id = data.id
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.isPrivate = data.isPrivate
    this.creatorId = data.creatorId
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.creator = data.creator ? new Profile(data.creator) : null
  }
}